### PR TITLE
Add notification when loading ACL

### PR DESCRIPTION
**Pull request by jcantrill**
@jimmidyson @ewolinetz  @sosiouxme  Replaced the thread.sleep with notification when ACL is loaded .  Please review

### CODE REVIEW

**Comment by jimmidyson** [comment]
Needs a rebase

**Comment by jcantrill** [comment]
@jimmidyson @ewolinetz @sosiouxme rebased

File: src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java [hunk 1]
```diff
@@ -56,28 +70,44 @@
...
+	@Override
 	public void process(RestRequest request, RestChannel channel, RestFilterChain chain) throws Exception {
 		try {
-			logger.debug("Handling Request in {}...", this.getClass().getSimpleName());
 			final String user = getUser(request);
 			final String token = getBearerToken(request);
-			logger.debug("Evaluating request for user '{}' with a {} token", user,
-					(StringUtils.isNotEmpty(token) ? "non-empty" : "empty"));
-			logger.debug("Cache has user: {}", cache.hasUser(user));
+			if(logger.isDebugEnabled()){
```
Comment (ewolinetz) [inline src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java]: We should probably wrap the rest of our debug log statements like this one

File: src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java [hunk 2]
```diff
@@ -156,15 +185,20 @@ private void write(Client esClient, SearchGuardACL acl) throws JsonProcessingExc
...
+			.setDoc(mapper.writeValueAsBytes(acl))
+			.setRefresh(true).request();
+		request.putInContext(OS_ES_REQ_ID, ACL_FILTER_ID);
+		esClient.update(request).actionGet();
+		lock.lock();
+		try{
+			logger.debug("Waiting up to {} ms. to be notified that SearchGuard has refreshed the ACLs", aclSyncDelay);
+			syncing.await(aclSyncDelay, TimeUnit.MILLISECONDS);
+		}catch(InterruptedException e){
+			logger.error("Error why awaiting notification of ACL load by SearchGuard", e);
```
Comment (ewolinetz) [inline src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java]: Should this read 'Error while awaiting' ?

File: src/main/java/io/fabric8/elasticsearch/plugin/ConfigurationSettings.java [hunk 3]
```diff
@@ -17,21 +17,25 @@
...
 	 */
-	public static final String OPENSHIFT_ES_ACL_DELAY_IN_MILLIS = "io.fabric8.elasticsearch.acl.sync_delay_millis";
-	public static final String OPENSHIFT_ES_USER_PROFILE_PREFIX = "io.fabric8.elasticsearch.acl.user_profile_prefix";
+	static final String OPENSHIFT_ES_ACL_DELAY_IN_MILLIS = "io.fabric8.elasticsearch.acl.sync_delay_millis";
+	static final String OPENSHIFT_ES_USER_PROFILE_PREFIX = "io.fabric8.elasticsearch.acl.user_profile_prefix";
 	
-	public static final String DEFAULT_AUTH_PROXY_HEADER = "X-Authenticated-User";
-	public static final String DEFAULT_SECURITY_CONFIG_INDEX = "searchguard";
-	public static final String DEFAULT_USER_PROFILE_PREFIX = ".kibana";
+	static final String DEFAULT_AUTH_PROXY_HEADER = "X-Authenticated-User";
```
Comment (ewolinetz) [inline src/main/java/io/fabric8/elasticsearch/plugin/ConfigurationSettings.java]: Per @sosiouxme comment on PR 9, should we change this to `X-Proxy-Remote-User` to match the auth proxy default?

**Comment by ewolinetz** [comment]
few nits but otherwise LGTM

**Comment by jimmidyson** [comment]
when you're ready to merge @ewolinetz @jcantrill @sosiouxme just comment with `[ merge ]` (without spaces) to get Jenkins to do the merge for you.

File: src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java (hunk 2 above)
Comment (jcantrill) [inline src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java]: fixed

**Comment by jcantrill** [comment]
@sosiouxme please confirm the change for the default user header

**Comment by jcantrill** [comment]
[merge]

**Pull request merged**